src/Microsoft.AspNetCore.Routing/Matchers/HttpMethodMatcherPolicy.cs

```diff
@@ -178,5 +291,58 @@ protected override int CompareMetadata(IHttpMethodMetadata x, IHttpMethodMetadat
                     y?.HttpMethods.Count > 0 ? y : null);
             }
         }
+
+        internal readonly struct EdgeKey : IEquatable<EdgeKey>, IComparable<EdgeKey>, IComparable
```

Comment:
I still think it's worth keeping as a struct since I use them as keys. There will be cases where I'm trying to look up something in the dictionary, and it's already there, so there's no need to allocate.